util/len-caching-lock/src/lib.rs

```diff
@@ -66,15 +62,15 @@ impl<T> Len for LinkedList<T> {
 	fn len(&self) -> usize { LinkedList::len(self) }
 }
 
-impl<K: Eq + Hash, V> Len for HashMap<K, V> {
+impl<K: Eq + Hash, V, S: std::hash::BuildHasher> Len for HashMap<K, V, S> {
```

Comment:
question: does this make it more generic instead of using https://doc.rust-lang.org/std/collections/hash_map/struct.RandomState.html?

Reply:
that's what the compiler told me to do
```
warning: impl for `HashMap` should be generalized over different hashers
  --> util/len-caching-lock/src/lib.rs:69:31
   |
69 | impl<K: Eq + Hash, V> Len for HashMap<K, V> {
   |                               ^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::implicit_hasher)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#implicit_hasher
help: consider adding a type parameter
   |
69 | impl<K: Eq + Hash, V, S: ::std::hash::BuildHasher> Len for HashMap<K, V, S> {
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^         ^^^^^^^^^^^^^^^^
```
:man_shrugging:

Reply:
Ok, yeah I think what I wrote is correct because https://doc.rust-lang.org/src/std/collections/hash/map.rs.html#202, uses a default type parameter. Then you can't use your custom Hasher, https://rust-lang.github.io/rust-clippy/master/#implicit_hasher